pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -127,62 +126,41 @@ public static Expression createNewLiteralExpression() {
     return expression;
   }
 
-  public static Expression getLiteralExpression(String value) {
+  public static Expression getLiteralExpression(long value) {
     Expression expression = createNewLiteralExpression();
-    expression.getLiteral().setStringValue(value);
+    expression.getLiteral().setLongValue(value);
     return expression;
   }
 
-  public static Expression getLiteralExpression(byte[] value) {
+  public static Expression getLiteralExpression(double value) {
     Expression expression = createNewLiteralExpression();
-    expression.getLiteral().setStringValue(BytesUtils.toHexString(value));
+    expression.getLiteral().setDoubleValue(value);
     return expression;
   }
 
-  public static Expression getLiteralExpression(Integer value) {
-    return getLiteralExpression(value.longValue());
-  }
-
-  public static Expression getLiteralExpression(Long value) {
+  public static Expression getLiteralExpression(String value) {
     Expression expression = createNewLiteralExpression();
-    expression.getLiteral().setLongValue(value);
+    expression.getLiteral().setStringValue(value);
     return expression;
   }
 
-  public static Expression getLiteralExpression(Float value) {
-    return getLiteralExpression(value.doubleValue());
-  }
-
-  public static Expression getLiteralExpression(Double value) {
+  public static Expression getLiteralExpression(byte[] value) {
     Expression expression = createNewLiteralExpression();
-    expression.getLiteral().setDoubleValue(value);
+    expression.getLiteral().setStringValue(BytesUtils.toHexString(value));
     return expression;
   }
 
   public static Expression getLiteralExpression(Object object) {
-    if (object instanceof Integer) {
-      return RequestUtils.getLiteralExpression((Integer) object);
-    }
-    if (object instanceof Long) {
-      return RequestUtils.getLiteralExpression((Long) object);
-    }
-    if (object instanceof Float) {
-      return RequestUtils.getLiteralExpression((Float) object);
-    }
-    if (object instanceof Double) {
-      return RequestUtils.getLiteralExpression((Double) object);
-    }
-    if (object instanceof String) {
-      return RequestUtils.getLiteralExpression((String) object);
+    if (object instanceof Integer || object instanceof Long) {
+      return RequestUtils.getLiteralExpression(((Number) object).longValue());
     }
-    if (object instanceof SqlLiteral) {
```

Comment:
We need to handle the case for SqlLiteral right ?

Reply:
No we don't. We have a separate method for query parsing. This one is for creating a literal expression from a constant value.